imap.json

```diff
@@ -27,31 +29,70 @@
             "order": 0,
             "required": true
         },
-        "ph": {
-            "data_type": "ph",
-            "order": 1
+        "auth_type": {
+            "data_type": "string",
+            "order": 1,
+            "description": "Authentication Mechanism to Use",
+            "value_list": [
+                "Basic",
+                "OAuth"
+            ],
+            "default": "Basic"
         },
         "username": {
             "data_type": "string",
             "description": "Username",
-            "order": 2
+            "order": 2,
+            "required": true
         },
         "password": {
             "data_type": "password",
             "description": "Password",
             "order": 3
         },
+        "client_id": {
+            "data_type": "string",
+            "description": "Client ID",
+            "order": 4
+        },
+        "client_secret": {
+            "data_type": "password",
+            "description": "Client Secret",
+            "order": 5
+        },
+        "auth_url": {
+            "data_type": "string",
+            "description": "Authorization URL",
+            "order": 6,
+            "default": "https://accounts.google.com/o/oauth2/auth"
+        },
+        "token_url": {
+            "data_type": "string",
+            "description": "Token URL",
+            "order": 7,
+            "default": "https://oauth2.googleapis.com/token"
+        },
+        "scopes": {
+            "description": "API Scope (JSON formatted list)",
+            "data_type": "string",
+            "default": "[\"https://mail.google.com/\"]",
+            "order": 8
+        },
+        "ph": {
```

Comment:
Can these new fields work with any other types of authentication (e.g. EWS Exchange)?

Reply:
For now, we only tested for the Gmail mailbox, because the [ticket](https://splunk.atlassian.net/browse/PAPP-9184) was created for Gmail. In the future, we can extend support for other mailboxes.

We have mentioned this limitation in the document [here](https://github.com/splunk-soar-connectors/imap/blob/cf3d6df510ea2f79fdffd975c73fa9ed90f8aecf/readme.html#L83).

I will rename these fields as per the discussion on slack.